common/src/main/java/io/netty/util/NetUtil.java

```diff
@@ -960,14 +960,26 @@ public static String toSocketAddressString(InetSocketAddress addr) {
         return sb.append(':').append(port).toString();
     }
 
-    private static StringBuilder newSocketAddressStringBuilder(String hostString, String port, boolean ipv4) {
+    /**
+     * Returns the {@link String} representation of a host port combo.
+     */
+    public static String toSocketAddressString(String host, int port) {
+        return newSocketAddressStringBuilder(
+                host, String.valueOf(port), isValidIpV4Address(host)).append(port).toString();
```

Comment:
int to string conversion may be reduced:
```java
String portStr = String.valueOf(port);
return newSocketAddressStringBuilder(host, portStr, isValidIpV4Address(host))
    .append(portStr)
    .toString();
```
Also `newSocketAddressStringBuilder()` can take only the length of portStr..